hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/AsyncSnapshotWriterImpl.java

```diff
@@ -242,19 +245,28 @@ private boolean putAsyncToMap(int partitionId, Supplier<Data> dataSupplier) {
     }
 
     /**
-     * Flush all partitions.
+     * Flush all partitions and reset current map. No further items can be
+     * offered until new snapshot is seen in {@link #snapshotContext}.
      *
      * @return {@code true} on success, {@code false} if we weren't able to
      * flush some partitions due to the limit on number of parallel async ops.
      */
     @Override
     @CheckReturnValue
-    public boolean flush() {
+    public boolean flushAndReset() {
         for (int i = 0; i < buffers.length; i++) {
```

Comment:
I think we should also return false here if map name is not set, since flush() can be called without any data.

Reply:
No, `flush` returns false if you need to call flush again. If no map was set, we're fine returning true.

Reply:
It'll be required if we want to prevent snapshot from completing before map is set.

Reply:
True, didn't realize it. Fixed.